examples/download_version.rs

```diff
@@ -1,12 +1,98 @@
-use std::env;
+use std::{
+    env,
+    io::Stdout,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, Mutex,
+    },
+};
 
-use mc_downloader::ClientDownloader;
+use mc_downloader::prelude::{ClientDownloader, DownloadVersion, Reporter};
+use pbr::ProgressBar;
+
+struct ProgressTrack {
+    curr: AtomicU64,
+    total: AtomicU64,
+    pb: Mutex<ProgressBar<Stdout>>,
+}
 
 fn main() {
-    let dir = env::current_dir().unwrap();
-    let target = format!("{}{}", dir.to_str().unwrap().to_string(), "\\.minecraft");
-    let downloader: ClientDownloader = ClientDownloader::new();
-    downloader
-        .download_version("1.19.3".to_string(), target)
-        .unwrap();
+    println!("Start Minecraft Downloader");
+    let mut dir = env::current_dir().unwrap();
+    dir.push(".minecraft");
+    match ClientDownloader::new() {
+        Ok(downloader) => {
+            println!("Start Download Minecraft 1.19.3 version");
```

Comment:
Don't hardcode it to a specific Minecraft version.

Reply:
I think that like the case above, since this is an example of how the library is used it should not really affect, but I think I see an opportunity to implement the example of how to extract the versions, even so I think this can be for another example or if you insist, I can fix it